### DISCUSSION TITLE
Tap maintenance and Homebrew development

### DISCUSSION THREAD

**Discussion by fugitiveMedia** [Homebrew/discussions#4969] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.24-12-gc32bd1c
ORIGIN: https://github.com/Homebrew/brew
HEAD: c32bd1c7cc0c3b1b914845bddfeda53f4d877a3f
Last commit: 2 days ago
Core tap JSON: 09 Dec 23:47 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.wTS5Cp77TI/org.xquartz:0
HOMEBREW_EDITOR: /usr/bin/vim
HOMEBREW_FORCE_BREWED_CA_CERTIFICATES: set
HOMEBREW_FORCE_BREWED_CURL: set
HOMEBREW_FORCE_BREWED_GIT: set
HOMEBREW_FORCE_VENDOR_RUBY: set
HOMEBREW_GIT_EMAIL: fugitivemedia@gmail.com
HOMEBREW_GIT_NAME: fugitivemedia
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 4
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: quad-core 64-bit haswell
Clang: 14.0.0 build 1400
Git: 2.43.0 => /usr/local/opt/git/bin/git
Curl: 8.5.0 => /usr/local/opt/curl/bin/curl
macOS: 12.7.1-x86_64
CLT: 14.2.0.0.1.1668646533
```

### Output of `brew doctor`

```shell
Warning: Putting non-prefixed coreutils in your path can cause GMP builds to fail.
[1]    22086 exit 1     brew doctor
```

### Description of issue

Currently spidermonkey and rsyncui are being blocked from upgraded on my Monterey OS computer due to OS versions. I am very familiar with XCode versions restriction which make sense(even if they are insane, when I have 4 macs of differing age that can't get the version you demand:). I'm running 3 intel macs and have been running Homebrew since you started(before that MacPorts which seems to be keeping up to today:). Your structure of installs is better to control most of the time. I think (just a developer who has worked in developing since the late 80's) Wouldn't it be a more controllable if you required installed packages by version numbers for dependence and everything was a versioned install. That way an install from my issue could be installed based on appropriate dependencies and scale back versions if needed. The definition of a formula as the newest hinders your system more than helps it. That change could have helped when you cut archived systems! An adaptive versioning system would help break a bunch of issues you have.

**Comment by SMillerDev** [comment]
> The definition of a formula as the newest hinders your system more than helps it. That change could have helped when you cut archived systems! An adaptive versioning system would help break a bunch of issues you have.

Technically, that may be true. But for this we would need a very significant increase in CI time and infrastructure along with an even larger increase in people willing to maintain Homebrew. And especially that last one is not something feasible to bet on.
